lib/ceremony-dropdown.tsx

```diff
@@ -0,0 +1,108 @@
+import { Ceremony } from "@/types/types";
+import { DropdownMenuRadioItem } from "@/components/ui/dropdown-menu";
+import { Badge } from "@/components/ui/badge";
+import { useState, useEffect } from "react";
+import { getCeremonyList, getCeremonyPassports } from "./get-passport-data";
+
+
+export default function CeremonyDropdown() {
+    const [ceremonyList, setCeremonyList] = useState<Ceremony[]>([]);
+    const [cLoading, setCLoading] = useState(true);
+
+    useEffect(() => {
+        const fetchCeremonies = async () => {
+            setCLoading(true);
+            try {
+                const ceremonies = await getCeremonyList();
+                setCeremonyList(ceremonies || []);
+            } catch (error) {
+                setCeremonyList([]);
+            } finally {
+                setCLoading(false);
+            }
+        };
+
+        fetchCeremonies();
+    }, []);
```

Comment:
If the goal is to fetch this data when the page loads, you should instead use `swr` or fetch the data in a server component and pass it down to the client, IMO. Data fetching in useEffect works but it's not very performant